invoice.go

```diff
@@ -195,9 +211,9 @@ type Invoice struct {
 	AttemptCount                 int64                    `json:"attempt_count"`
 	Attempted                    bool                     `json:"attempted"`
 	AutoAdvance                  bool                     `json:"auto_advance"`
-	Billing                      InvoiceBilling           `json:"billing"`
 	BillingReason                InvoiceBillingReason     `json:"billing_reason"`
 	Charge                       *Charge                  `json:"charge"`
+	CollectionMethod             *InvoiceCollectionMethod `json:"collection_method"`
```

Comment:
Any reason to move to a pointer here?
I believe the API is always providing this field and this makes the comparison with `InvoiceCollectionMethodSendInvoice` or  `InvoiceCollectionMethodChargeAutomatically` less smooth.
If you confirm this should have stayed a value I can send a PR.

Reply:
@remi-stripe makes sense.
Though, what about the change on `Billing`? It was a BC break for people upgrading (that's actually how I spot this issue in the first place).
Do you want to revert this in current version or wait the same way?

Reply:
@tucksaun which change on `Billing`? We follow semantic versioning so we make all breaking changes in major versions.

Reply:
@remi-stripe https://github.com/stripe/stripe-go/pull/880/files#diff-7d99b0ab199591cd498ef67d8e82c4e1L198 became a pointer (https://github.com/stripe/stripe-go/pull/880/files#diff-7d99b0ab199591cd498ef67d8e82c4e1R266) in this PR and released as of `v61.7.0`

Reply:
@tucksaun Oh wow, I'm sorry about that. This shouldn't have happened. I was just marking the property as deprecated for a future deprecation and the type should definitely not have changed at that point. That also explains why the other type is a pointer when it should not, because I must have set both incorrectly at the same time.